cluster/src/main/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializer.java

```diff
@@ -593,75 +815,385 @@ public void close() {
+      logDataFile = logDataFileList.get(index);
+      logIndexFile = logIndexFileList.get(index);
+      Files.delete(logDataFile.toPath());
+      Files.delete(logIndexFile.toPath());
+      logDataFileList.remove(index);
+      logIndexFileList.remove(index);
+      logger.debug("delete date file={}, index file={}", logDataFile.getAbsoluteFile(),
+          logIndexFile.getAbsoluteFile());
+    } catch (IOException e) {
+      logger.error("delete file failed, index={}, data file={}, index file={}", index,
+          logDataFile == null ? null : logDataFile.getAbsoluteFile(),
+          logIndexFile == null ? null : logIndexFile.getAbsoluteFile());
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  /**
+   * The file name structure is as follows： {startLogIndex}-{endLogIndex}-{version}-data)
+   *
+   * @param file1 File to compare
+   * @param file2 File to compare
    */
-  private void adjustNextThreshold() {
-    if (!logFileList.isEmpty()) {
-      maxRemovedLogSize = logFileList.get(0).length();
+  private int comparePersistLogFileName(File file1, File file2) {
+    String[] items1 = file1.getName().split(FILE_NAME_SEPARATOR);
+    String[] items2 = file2.getName().split(FILE_NAME_SEPARATOR);
+    if (items1.length != FILE_NAME_PART_LENGTH || items2.length != FILE_NAME_PART_LENGTH) {
+      logger.error(
+          "file1={}, file2={} name should be in the following format: startLogIndex-endLogIndex-version-data",
+          file1.getAbsoluteFile(), file2.getAbsoluteFile());
+    }
+    long startLogIndex1 = Long.parseLong(items1[0]);
+    long startLogIndex2 = Long.parseLong(items2[0]);
+    int res = Long.compare(startLogIndex1, startLogIndex2);
+    if (res == 0) {
+      return Long.compare(Long.parseLong(items1[1]), Long.parseLong(items2[1]));
     }
+    return res;
   }
 
   /**
-   * actually delete the data file which only contains removed data
+   * @param startIndex the log start index
+   * @param endIndex   the log end index
+   * @return the raft log which index between [startIndex, endIndex] or empty if not found
    */
-  private void actuallyDeleteFile() {
-    Iterator<File> logFileIterator = logFileList.iterator();
-    while (logFileIterator.hasNext()) {
-      File logFile = logFileIterator.next();
-      if (logger.isDebugEnabled()) {
-        logger.debug("Examining file for removal, file: {}, len: {}, removedLogSize: {}", logFile
-            , logFile.length(), removedLogSize);
-      }
-      if (logFile.length() > removedLogSize) {
-        break;
-      }
-
-      logger.info("Removing a log file {}, len: {}, removedLogSize: {}", logFile,
-          logFile.length(), removedLogSize);
-      removedLogSize -= logFile.length();
-      // if system down before delete, we can use this to delete file during recovery
-      minAvailableVersion = getFileVersion(logFile);
-      serializeMeta(meta);
+  @Override
+  public List<Log> getLogs(long startIndex, long endIndex) {
```

Comment:
I think we should enforce a limit on this method to avoid out ot memory when the range is too long.

Reply:
I add one parameter to limit the max number of logs per fetch